docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,552 @@
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,bash,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-avro-schema-quickstart \
+    -DclassName="org.acme.kafka.MovieResource" \
+    -Dpath="/movies" \
+    -Dextensions="resteasy-reactive,resteasy-reactive-jackson,rest-client,smallrye-reactive-messaging-kafka,avro"
+cd kafka-avro-schema-quickstart
+----
+
+Please note that even thought our application will not explicitly use a REST client, it is used
+under the hood by the Apicurio registry and, at the moment it is required as a dependency.
+
+Additionally, we need a serializer and a deserializer for Avro, specific to the schema registry we are going to use.
+Apicurio, which we use in this guide, has them in the following dependency:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.apicurio</groupId>
+    <artifactId>apicurio-registry-utils-serde</artifactId>
+    <version>1.2.2.Final</version>
+    <exclusions>
+        <exclusion>
+            <groupId>org.jboss.spec.javax.interceptor</groupId>
+            <artifactId>jboss-interceptors-api_1.2_spec</artifactId>
+        </exclusion>
+    </exclusions>
+</dependency>
+----
+
+
+== Avro schema
+Let's create a file called `src/main/avro/movie.avsc` with the schema for our record (Kafka message):
+[source,javascript]
+----
+{
+  "namespace": "org.acme.kafka.quarkus",
+  "type": "record",
+  "name": "Movie",
+  "fields": [
+    {
+      "name": "title",
+      "type": "string"
+    },
+    {
+      "name": "year",
+      "type": "int"
+    }
+  ]
+}
+----
+
+If you build the project with `mvn compile` , the `movies.avsc` will get compiled to a `Movie.java` file
+placed in the `target/generated-sources/avsc` directory.
+
+If you run the project with `mvn compile quarkus:dev`, the changes you do to the schema file will be
+automatically applied to the generated Java files.
+
+== The `Movie` producer
+Having defined the record, we can now jump to implementing the `MovieResource`.
+
+Let's open the `MovieResource`, inject an `Emitter` of `Movie` DTO and implement a `@POST` method
+that consumes `Movie` and sends it through the `Emitter`:
+
+[source,java]
+----
+package org.acme.kafka;
+
+import org.acme.kafka.quarkus.Movie;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.jboss.logging.Logger;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/movies")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class MovieResource {
+
+    private static final Logger LOGGER = Logger.getLogger(MovieResource.class);
+
+    @Channel("movies") Emitter<Movie> emitter;
```

Comment:
Isn't `@Inject` missing? Maybe it's not needed and I never realized it.

Reply:
This is a non-standard extension to CDI in Quarkus. If you use a qualifier, you don't have to add `@Inject`.